addons/binding/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/handler/ICloudDeviceHandler.java

```diff
@@ -161,15 +166,15 @@ private DeviceInformation getDeviceInformationRecord(List<DeviceInformation> dev
         return null;
     }
 
-    private State getLastLocationUpdateDateTimeState(DeviceInformation deviceInformationRecord) {
+    private State getLastLocationUpdateDateTimeState(ICloudDeviceInformation deviceInformationRecord) {
         State dateTime = UnDefType.UNDEF;
 
         if (deviceInformationRecord.getLocation().getTimeStamp() > 0) {
-            Date javaDate = new Date(deviceInformationRecord.getLocation().getTimeStamp());
-            SimpleDateFormat javaDateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssz");
-            String lastUpdate = javaDateFormat.format(javaDate);
-
-            dateTime = new DateTimeType(lastUpdate);
+            Date date = new Date(deviceInformationRecord.getLocation().getTimeStamp());
+            Calendar calendar = new GregorianCalendar();
```

Comment:
The calendar does not seem to be involved in the calculation anymore?

Reply:
... done.